### DISCUSSION TITLE
Reduce website & docs complexity

### DISCUSSION THREAD

**Discussion by dominikriemer** [apache/streampipes#1731] [Dev]
Hi,
currently, we use three different frameworks to host our website:
* The website itself using harp.js
* The documentation using Docusaurus v1
* The Python docs using markdown

The reason why we use two frameworks for website & docs was that Docusaurus v1, which we used at this time, did not provide a lot of customization features in terms of design, e.g., building a nice-looking home page.

Docusaurus v2 is already available for some time and can be completely customized with React under the hood, and although it is some substantial work, I suggest to merge the website and docs and then serve the main content from Docusaurus. This should ease website maintenance and allows users to faster access the documentation instead of navigating to a subpage. We would keep the Python docs in Markdown since it's automatically built but would also be able to show the Python support more prominently in the navigation bar.

I created a branch in the website repo (https://github.com/apache/streampipes-website/tree/improved-website) to check the feasibility and it roughly works - I'll continue to migrate the home page and other pages.

What do you think? Do we have any requirements or changes to the main structure we want to have?

**Comment by tenthe** [comment]
+1 from my side. I think it makes sense to simplify the setup

**Comment by bossenti** [comment]
@dominikriemer thanks  a lot for starting this initiative and for starting the migration process
Your proposal sounds good to me, so I'm +1 as well

One minor correction, the Python library uses [mkdocs-material](https://squidfunk.github.io/mkdocs-material/) for it's documentation
Let's see how happy we are with docusaurus v2
If we fully like it my personal suggestion would be to investigate whether we can use it for the Python docs as well
This would simplify things a lot I guess

**Comment by bossenti** [comment]
If the migration is done successfully we might also reconsider to integrate the homepage in our main repo as well see #965

**Comment by dominikriemer** [comment]
Hi,
I've created a PR at https://github.com/apache/streampipes-website/pull/93

I improved the look and feel of the homepage as well - it would be great if you could have a look and give feedback, especially in terms of the whole message - does the main page correctly explain everything we do?

You can run the website by navigating to the `website-v2` folder and executung `npm install` and then `npm start`.

cc @tenthe @bossenti @RobertIndie @obermeier

**Comment by dominikriemer** [comment]
The new website is online, please have a look and let me know in case you have any feedback!

**Reply by bossenti** [reply]
It is amazing 🤩 really love it 
I'm looking forward to improving all the small & fine details and polish our docs 🙂
